src/threading.c

```diff
@@ -960,6 +960,32 @@ JL_DLLEXPORT int jl_alignment(size_t sz)
     return jl_gc_alignment(sz);
 }
 
+JL_DLLEXPORT int jl_getaffinity(int16_t tid, char *mask, int cpumasksize) {
+    int nthreads = jl_atomic_load_acquire(&jl_n_threads);
+    if (tid < 0 || tid >= nthreads)
+        jl_error("invalid tid");
+
+    // TODO: relaxed or acquire here?
+    jl_ptls_t *tls = jl_atomic_load_relaxed(&jl_all_tls_states);
+    uv_thread_t uvtid = tls[tid]->system_id;
```

Comment:
To follow the naming convention in the rest of the code

```suggestion
    jl_ptls_t *ptls2 = jl_atomic_load_relaxed(&jl_all_tls_states)[tid];
    uv_thread_t uvtid = ptls2->system_id;
```

Reply:
It seems like this should require a lock, since `system_id` is only legal if the thread is alive, and otherwise `ptls2` might be NULL

Reply:
@vtjnash, not sure I can implement this. Would you mind making a commit suggestion?

Reply:
Good question. We seem to have far too many different locks for this, which aren't exactly coordinated.
Some of the code uses one of these 4 locks
https://github.com/JuliaLang/julia/blob/3351e574e386b44ff361daa4ade8eb23270d951e/src/threading.c#L472-L485
Other code either generally fails to check
https://github.com/JuliaLang/julia/blob/3351e574e386b44ff361daa4ade8eb23270d951e/src/safepoint.c#L326-L328
which is perhaps assuming that the safepoint_lock is doing the synchronization (though it does not)

maybe we just need to punt on this question and put a TODO comment in the code in its place to use the correct lock here

Reply:
I've put a TODO comment in the code.